File: 155-min-stack/155-min-stack.ts
class MinStack {
    public minList: number[];
    public list: number[];
    constructor() {
      this.minList = [];
      this.list = [];
    }
    push(val: number): void {
      this.list.push(val);
      if (this.minList.length === 0 || this.minList[this.minList.length - 1] >= val) {
        this.minList.push(val);
      }
    }
    pop(): void {
      const val = this.list.pop();
      if (this.minList[this.minList.length - 1] === val) {
        this.minList.pop();
      }
    }
    top(): number {
      return this.list[this.list.length - 1];
    }
    getMin(): number {
      return this.minList[this.minList.length - 1];
    }
}
/**
 * Your MinStack object will be instantiated and called as such:
 * var obj = new MinStack()
 * obj.push(val)
 * obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.getMin()
 */

File: 155-min-stack/155-min-stack.py
class MinStack:

    def __init__(self):
        self.stk = []
        self.minlist = []
        

    def push(self, val: int) -> None:
        self.stk.append(val)
        if len(self.minlist) == 0 or self.minlist[-1] >= val:
            self.minlist.append(val)
        

    def pop(self) -> None:
        item = self.stk.pop()
        if item == self.minlist[-1]:
            self.minlist.pop()
        

    def top(self) -> int:
        return self.stk[-1]
        

    def getMin(self) -> int:
        return self.minlist[-1]
        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()